File: SDK-64bit/doc/_downloads/almotion_getlimits.py
import sys
from naoqi import ALProxy

if (len(sys.argv) < 2):
    print "Usage: 'python motion_getlimits.py IP [PORT]'"
    sys.exit(1)

IP = sys.argv[1]
PORT = 9559
if (len(sys.argv) > 2):
    PORT = sys.argv[2]
try:
    proxy = ALProxy("ALMotion", IP, PORT)
except Exception,e:
    print "Could not create proxy to ALMotion"
    print "Error was: ",e
    sys.exit(1)

# Example showing how to get the limits for the whole body 
name = "Body" 
limits = proxy.getLimits(name) 
jointNames = proxy.getJointNames(name) 
for i in range(0,len(limits)): 
  print jointNames[i] + ":"
  print "MinAngle", limits[i][0], "MaxAngle", limits[i][1],"MaxVelocity", limits[i][2]

File: SDK-64bit/doc/_downloads/almotion_getlimits.cpp
#include <iostream>
#include <alproxies/almotionproxy.h>

int main(int argc, char **argv)
{
  if (argc < 2) {
    std::cerr << "Usage: motion_getlimits pIp"
              << std::endl;
    return 1;
  }
  const std::string pIp = argv[1];

  AL::ALMotionProxy motion(pIp);

  // Example showing how to get the limits for the whole body
  std::string name = "Body";
  std::vector<std::string> jointNames = motion.getJointNames(name);
  AL::ALValue limits = motion.getLimits(name);
  for (unsigned int i=0; i<limits.getSize(); i++)
  {
    std::cout << " Joint name " << jointNames[i]
              << " MinAngle " << limits[i][0] << " rad"
              << " MaxAngle " << limits[i][1] << " rad"
              << " MaxChange " << limits[i][2] << " rad.s-1"
              << std::endl;
  }

  return 0;
}
